### PR TITLE
Simple Github Metric Support

### PR DESCRIPTION
**Pull request by ngtvspc**
Adds two main features:
- support for required env variables
- github collect-metrics module

Notes to reviewer:
Not happy about the DAL extensions I made, would like to hear any suggestions of how to better structure it.
Any comments on the use of curl would be appreciated too. Might be some benefit to using another lib like requests?

### CODE REVIEW

**Comment by ngtvspc** [comment]
Sample Output:
```
{"metric_name": "gh", "project": "ubiquity", "pull_date": "2019-08-10T23:43:22", "result": {}, "~ms": 0.23}
{"metric_name": "gh", "project": "voltron", "pull_date": "2019-08-10T23:43:23", "result": {"forks": 327, "has_wiki": true, "license": "MIT License", "open_issues": 61, "size": 1673, "stars": 4892, "watchers": 161}, "~ms": 405.79}
{"metric_name": "gh", "project": "guake", "pull_date": "2019-08-10T23:43:23", "result": {"forks": 501, "has_wiki": false, "license": "GNU General Public License v2.0", "open_issues": 353, "size": 6940, "stars": 3000, "watchers": 146}, "~ms": 304.23}
```

**Comment by mahmoud** [comment]
Empty dict is good, but yeah, re: curl, I think stdlib should be fine actually: https://github.com/mahmoud/zerover/blob/810c8bf750b7996110a27cb7388f90c5d00ab92f/tools/gen_projects_json.py#L44

**Comment by ngtvspc** [comment]
Okeydokey, changed to use urllib instead of curl, wasn't too difficult.

File: apatite/apatite/cli.py [hunk 1]
```diff
@@ -381,22 +381,41 @@ def _get_all_metric_mods():
...
     return ret
 
+# TODO _check_required_env_vars and _check_required_cmds have similar
+# structures, maybe refactor to more generic
+def _check_required_env_vars(metric_mod):
+    UNSET = object()
+    missing_env_vars = []
+    required_env_vars = getattr(metric_mod, 'required_env_vars', [])
+    for env_var in required_env_vars:
+        var_value = os.getenv(env_var, UNSET)
```
Comment (mahmoud) [inline apatite/apatite/cli.py]: getenv can only really return strings, so `None` (the default) is a fine sentinel to use.

File: apatite/apatite/metrics/gh.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import glom
+import json
+import os
+import urllib.request
+
+
+DETECT_CMD = 'curl'
+DETECT_ENV = 'GH_API_KEY'
+
+required_cmds = {DETECT_CMD: 'install via your package manager'}
```
Comment (mahmoud) [inline apatite/apatite/metrics/gh.py]: curl is no longer required, so I'd drop required_cmds or leave it as an empty dict

**Comment by mahmoud** [comment]
Couple of comments above, but I'll clean those up. Great work, and thanks for thinking to include more than just stars/forks :)

**Pull request merged**